### ISSUE TITLE
uncaught_exception: false positive (deps)

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#1813]
If the used exception is a child of the exception defined in the `catch` statement, the rule gives a false positive.

The example is a bit long, sorry

```abap
REPORT zfoobar.
CLASS lcl_alv DEFINITION.
  PUBLIC SECTION.
    METHODS set_text
      IMPORTING
        io_columns TYPE REF TO cl_salv_columns_table
        iv_column  TYPE lvc_fname
        iv_text    TYPE string
      RAISING
        cx_salv_not_found.
    METHODS show_alv.
  PRIVATE SECTION.
    DATA mt_alv TYPE TABLE OF t100.
ENDCLASS.

CLASS lcl_alv IMPLEMENTATION.

  METHOD set_text.
    DATA: lo_column      TYPE REF TO cl_salv_column,
          lv_long_text   TYPE scrtext_l.

    lo_column = io_columns->get_column( iv_column ).
    lv_long_text   = iv_text.
    lo_column->set_long_text( lv_long_text ).
  ENDMETHOD.

  METHOD show_alv.
    DATA: lx_error   TYPE REF TO cx_salv_error,
          lo_columns TYPE REF TO cl_salv_columns_table,
          lo_alv     TYPE REF TO cl_salv_table.

    TRY.
        cl_salv_table=>factory(
          IMPORTING
            r_salv_table = lo_alv
          CHANGING
            t_table      = mt_alv ).

        lo_columns = lo_alv->get_columns( ).

        set_text( io_columns = lo_columns
                  iv_column  = |TEST|
                  iv_text    = |Text| ).

        lo_alv->display( ).

      CATCH cx_salv_error INTO lx_error.
* cx_salv_error is a parent of cx_salv_not_found
    ENDTRY.
  ENDMETHOD.
ENDCLASS.
```

/zfoobar.prog.abap[41, 9]: Uncaught exception cx_salv_not_found(uncaught_exception)

**Comment by larshp** [comment]
reduced to below code

In this scenario, the linter does not know about the inheritance, to fix this, add the exception classes in https://github.com/abaplint/deps or a custom deps repo, I think that should work

```abap
CLASS lcl_alv DEFINITION.
  PUBLIC SECTION.
    METHODS set_text
      RAISING
        cx_salv_not_found.
    METHODS show_alv.
ENDCLASS.

CLASS lcl_alv IMPLEMENTATION.

  METHOD set_text.
    RETURN.
  ENDMETHOD.

  METHOD show_alv.
    DATA: lx_error TYPE REF TO cx_salv_error.

    TRY.
        set_text( ).
      CATCH cx_salv_error INTO lx_error.
* cx_salv_error is a parent of cx_salv_not_found
    ENDTRY.
  ENDMETHOD.
ENDCLASS.
```

**Comment by larshp** [comment]
@mbtools try again, added in https://github.com/abaplint/deps/pull/41

**Comment by mbtools** [comment]
needs one more cx
https://github.com/abaplint/deps/pull/42

**Comment by larshp** [comment]
merged

**Comment by larshp** [comment]
@mbtools try again

**Comment by mbtools** [comment]
the issues is still there in abaplint 2.70.0

![image](https://user-images.githubusercontent.com/59966492/110517975-d7330300-810b-11eb-8a79-7a54129cdeca.png)

reproduce in playground with your reduced example above

**Comment by larshp** [comment]
@mbtools try again, https://github.com/abaplint/deps/pull/47 wrong filename in deps

**Comment by mbtools** [comment]
🏆 

![image](https://user-images.githubusercontent.com/59966492/110762988-af4db780-8251-11eb-859a-3c8e5ec8a140.png)
